cli/cmd/init.go

```diff
@@ -220,12 +220,47 @@ func validateStackNameAvailable(ctx context.Context, stackService *stackservice.
 			return nil
 		}
 
-		_, err := stackService.GetStack(ctx, stackName)
+		metas, err := stackService.CollectStackInfo(ctx, happyClient.HappyConfig.App(), true)
 		if err != nil {
+			return errors.Wrap(err, "unable to collect stack info")
+		}
+
+		for _, meta := range metas {
+			if meta.Stack == stackName {
+				if meta.AppName == happyClient.HappyConfig.App() {
+					return nil
+				}
+				return errors.Errorf("this stack exists, but in a different app ('%s')", meta.AppName)
+			}
+		}
+
+		return errors.Errorf("stack %s doesn't exist", stackName)
+	}
+}
+
+func validateStackNameGloballyAvailable(ctx context.Context, stackService *stackservice.StackService, stackName string, force bool) validation {
+	logrus.Debug("Scheduling validateStackNameAvailable()")
+	return func() error {
+		logrus.Debug("Running validateStackNameGloballyAvailable()")
+		if force {
 			return nil
 		}
 
-		return errors.Wrap(err, "the stack name is already taken")
+		metas, err := stackService.CollectStackInfo(ctx, happyClient.HappyConfig.App(), true)
+		if err != nil {
+			return errors.Wrap(err, "unable to collect stack info")
+		}
+
+		for _, meta := range metas {
+			if meta.Stack == stackName {
+				if meta.AppName == happyClient.HappyConfig.App() {
```

Comment:
```suggestion
			if meta.Stack == stackName && meta.AppName == happyClient.HappyConfig.App() {
				
```

can you make that fewer nested statements?